File: DSA/13_Graphs/adjacencyMatrix.py
matrix = []
# v = int(input("Enter the no of vertices: "))
with open("input.txt", "r") as f:
    data = f.read()
    data = data.split(" ")

rows = columns = int(data[0])

# Empty Matrix
for i in range(rows):
    temp = []
    for j in range(columns):
        temp.append(0)
    matrix.append(temp)

# edges = int(input("Enter the no of edges: "))
edges = int(data[1])
for i in range(2, edges + 2):
    # u = int(input(f"Enter the end vertices of edge:{i}: "))
    # v = int(input(f"Enter the end vertices of edge:{i}: "))
    u, v = data[i].split(",")
    u = int(u)
    v = int(v)
    matrix[u][v] = matrix[v][u] = 1

# Adjacency Matrix
for i in matrix:
    print(i)

File: DSA/13_Graphs/adjacencyMatrix.js
import fs from "fs"
let data = fs.readFileSync("input.txt", "utf-8");
data = data.split(" ");
let rows = parseInt(data[0]);
let cols = parseInt(data[0]);
let matrix = [];

for (let i = 0; i < rows; i++) {
    let row = [];
    for (let j = 0; j < cols; j++) {
        row.push(0);
    }
    matrix.push(row);
}
console.log(matrix);

let e = parseInt(data[1]);
for (let i = 2; i < e + 2; i++) {
    let edge = data[i].split(",");
    let x = parseInt(edge[0]);
    let y = parseInt(edge[1]);
    matrix[x][y] = matrix[y][x] = 1;
}
console.log(matrix);